src/main/java/lotto/domain/LottoNumbers.java

```diff
@@ -0,0 +1,57 @@
+package lotto.domain;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+public class LottoNumbers {
+
+    List<LottoNumber> lottoNumbers;
+
+    public LottoNumbers(final List<LottoNumber> lottoNumbers) {
+        this.lottoNumbers = lottoNumbers;
+    }
+
+    public List<LottoNumber> getLottoNumbers() {
+        return lottoNumbers;
+    }
+
+    public int countOfMatch(final LottoNumbers lottoNumbers) {
+        if (Objects.isNull(lottoNumbers)) {
+            return 0;
+        }
+
+        return this.countOfMatch(lottoNumbers.getLottoNumbers());
+    }
+
+
+    public int countOfMatch(final List<LottoNumber> lottoNumbers) {
+        if (Objects.isNull(lottoNumbers)) {
+            return 0;
+        }
+
+        return (int) lottoNumbers.stream()
+                                 .map(Optional::ofNullable)
+                                 .filter(Optional::isPresent)
+                                 .map(Optional::get)
+                                 .filter(this.lottoNumbers::contains)
+                                 .count();
```

Comment:
`Optional`을 체크하는 로직이 없어도 되지 않을까요?
```suggestion
        return (int) lottoNumbers.stream()
                                 .filter(this.lottoNumbers::contains)
                                 .count();
```

Reply:
반영하겠습니다~